mu/interface/panes.py

```diff
@@ -156,8 +169,14 @@ def __init__(self, serial, theme="day", parent=None):
         self.setUndoRedoEnabled(False)
         self.setContextMenuPolicy(Qt.CustomContextMenu)
         self.customContextMenuRequested.connect(self.context_menu)
+        # The following variable maintains the position where we know
+        # the device cursor is placed. It is initialized to the beginning
+        # of the QTextEdit (i.e. equal to the Qt cursor position)
+        self.device_cursor_position = self.textCursor().position()
         self.setObjectName("replpane")
         self.set_theme(theme)
+        self.unprocessed_input = b""  # used by process_bytes
+        self.decoder = codecs.getincrementaldecoder("utf8")()
```

Comment:
Oh, this is a great find in the standard library!
Does that mean that if an argument to the `decode` function call doesn't have all the bytes for a utf-8 char (so the last byte of the data array contains the first byte of a multi-byte characcter) it will keep state and concatenate incomplete characters into the next `decode` call?

Reply:
Yes, it decodes a bytestring in multiple steps and keeps a state of bytes that wasn't parsed between the steps. The explanation in the Python docs is a bit cryptic.